server/s3_object_metadata.cc

```diff
@@ -35,126 +37,124 @@
 
 extern struct m0_uint128 global_instance_id;
 
-void S3ObjectMetadata::initialize(bool ismultipart, std::string uploadid) {
-  json_parsing_error = false;
+S3ObjectMetadata::S3ObjectMetadata(const S3ObjectMetadata& src)
+    : S3ObjectMetadataCopyable(src), state(S3ObjectMetadataState::present) {
+
+  s3_log(S3_LOG_DEBUG, request_id, "Copy constructor");
+
+  bucket_name = request->get_bucket_name();
```

Comment:
if its copy constructor, why not read bucket name and object name from the passed object itself ?

Reply:
Partially agree. Quite strange copy constructor. I thought about separate function for updating partial copy. If you insist i will do this.

Reply:
No I do not insist, just a suggestion to make sure we are not doing this by mistake.

If this is conscious decision, then its fine. Make sure we write good UTs to catch any error in logic.